.circleci/config.yml

```diff
@@ -127,33 +132,33 @@ jobs:
           paths:
             - client/.happypack
 
-      - type: shell
-        name: Go Meta Linter
-        command: scripts/gometalinter.sh
-
+  go:
+    <<: *defaults
+    steps:
+      - restore_cache:
+          key: workspace-{{ .Environment.CIRCLE_WORKFLOW_WORKSPACE_ID }}
+      - run:
+          name: Go Meta Linter
+          command: scripts/gometalinter.sh
       - restore_cache:
           keys:
             - go-{{ .Branch }}-{{ .Revision }}
             - go-{{ .Branch }}
             - go-master
-
       - run:
           name: touch files restored from cache
           command: |
             timestamp=$(date --date="@$(($(date +%s) + 60))" +%m%d%H%M)
             find go/bin -exec touch {} -t $timestamp \;
             find go/pkg -exec touch {} -t $timestamp \;
-
       - run:
           name: touch changed go source files
           command: |
             timestamp=$(date --date="@$(($(date +%s) + 120))" +%m%d%H%M)
             base=origin/master
             head=$(git rev-parse HEAD)
             git diff --name-only $base...$head go/ | xargs --no-run-if-empty touch -t $timestamp
-
       - run: env GOGC=10 go/build.sh
-
       - save_cache:
           key: go-{{ .Branch }}-{{ .Revision }}-{{ epoch }}
```

Comment:
what about just storing {{ .Branch }} and not caring about other stuff ie revisions?